web/src/components/people/AdminView.tsx

```diff
@@ -39,42 +44,93 @@ export default function AdminView() {
     fetchPeople();
   }, []);
 
+  const updatePersonUI = (id: string, training?: TrainingRecord) => {
+    // get people object and find the person with the right id
+    const personToUpdate = people!.find((person) => person.user.id === id);
+
+    if (personToUpdate) {
+      if (training) {
+        if (!personToUpdate.training_record) {
+          personToUpdate.training_record = [];
+        }
+        personToUpdate.training_record.push(training);
+      }
+    }
+  };
+
   if (!people) return null;
 
+  const handleEditTrainingClick = (id: string) => {
+    setId(id);
+    setTrainingDialogOpen(true);
+  };
+
   return (
     <>
+      {trainingDialogOpen && (
+        <TrainingForm id={id} setTrainingDialogOpen={setTrainingDialogOpen} updatePersonUI={updatePersonUI} />
+      )}
+
       <table className={styles.table}>
         <thead>
           <tr>
             <th>User</th>
             <th>Roles</th>
             <th>Agreements</th>
+            <th>Training</th>
           </tr>
         </thead>
-        <tbody>
+        <tbody className={styles.tbody}>
           {people.map((person: Person) => (
             <tr key={person.user.id}>
               <td className={styles.user}>
                 {person.user.username} <small>{person.user.id}</small>
               </td>
-              <td className={styles.roles}>{person.roles}</td>
+              <td className={styles.roles}>{person.roles.join(", ")}</td>
               <td className={styles.agreements}>
                 {person.agreements.confirmed_agreements.map((agreement: ConfirmedAgreement) => (
                   <div key={agreement.agreement_type} className={styles.agreement}>
                     {agreement.agreement_type}
                     {agreement.confirmed_at ? (
-                      <span className={styles.confirmed}>
+                      <span className={styles.valid}>
```

Comment:
I've forgotten exactly how this works. Is it possible to have an agreement entry in the DB for a user and it not be valid?

Reply:
currently there is no expiry on the agreement